app/models/job_invocation_composer.rb

```diff
@@ -82,6 +82,14 @@ def triggering
       return trig unless trig.key?(:end_time) && trig[:end_time].keys == keys
       trig.merge(:end_time => Time.local(*trig[:end_time].values_at(*keys)))
     end
+
+    def targeting(targeting_params)
+      to_return = targeting_params.merge(:user_id => User.current.id)
+      if !to_return[:search_query].nil? && to_return[:search_query].empty?
+        to_return[:search_query] = "name != ''" # a dummy search query to match all hosts
```

Comment:
Could you put the this search to https://github.com/theforeman/foreman_remote_execution/pull/350/files#diff-5becfffb6454c87c9b39dd84ae3676d2R16, so that it only applies, when the job invocation page is accessed via the bulk action and the 'all hosts' is selected (behind a condition of `params.key?(:search) && params[:search].empty?`. Otherwise, this would be applied also in cases, where it would be confusing.